src/Athena/Controllers/api/CourseController.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Athena.Core.Repositories;
+using Athena.Core.Models;
+using Athena.Exceptions;
+using System.Net;
+
+namespace Athena.Controllers.api
+{
+    [Route("api/v1/[Controller]")]
+    public class CourseController : Controller
+    {
+        private readonly ICourseRepository courses;
+        private readonly IInstitutionRepository institutions;
+        private readonly IStudentRepository students;
+        private readonly IRequirementRepository requirements;
+
+        public CourseController(ICourseRepository coursesRepository, IInstitutionRepository institutionRepository, IStudentRepository studentRepository, IRequirementRepository requirementRepository)
+        {
+            courses = coursesRepository ?? throw new ArgumentNullException(nameof(coursesRepository));
+            institutions = institutionRepository ?? throw new ArgumentNullException(nameof(institutionRepository));
+            students = studentRepository ?? throw new ArgumentNullException(nameof(studentRepository));
+            requirements = requirementRepository ?? throw new ArgumentNullException(nameof(requirementRepository));
+        }
+
+        [HttpPost]
+        public async Task AddCourse([FromBody] Course course) => await courses.AddAsync(course);
+
+        [HttpGet("{id}")]
+        public async Task<Course> GetCourse(Guid id) => await courses.GetAsync(id);
```

Comment:
We should throw an ApiException with a `NotFound` status code if the repo returns a null